driver/src/main/scala/api/cursor.scala

```diff
@@ -510,7 +510,8 @@ object DefaultCursor {
         val max = if (maxDocs < 0) Int.MaxValue else maxDocs
         val toReturn = { // normalize the number of docs to return
           if (numberToReturn > 0 && numberToReturn <= max) numberToReturn
-          else max
+          else max - (
+            response.reply.numberReturned + response.reply.startingFrom)
```

Comment:
subjective but `response.reply.startingFrom + response.reply.numberReturned` would be more natural for me (offset+len rather than len+offset)